CppExport/src/exporter/CppExporter.cpp

```diff
@@ -34,74 +34,157 @@
+}
 
-	auto layout = layouter();
+QList<CodeComposite*> CppExporter::mergeUnits(QList<CodeUnit*>& units)
+{
+	QHash<QString, QString> mergeMap = Config::instance().dependencyUnitMergeMap();
 
-	DeclarationVisitorHeader headerVisitor;
-	auto dir = std::unique_ptr<Export::SourceDir>( headerVisitor.visitProject(project) );
-	auto map = Export::Exporter::exportToFileSystem(pathToProjectContainerDirectory + "/Headers", dir.get(), &layout);
-	exportMaps().insert(project, map);
+	QHash<QString, CodeComposite*> nameToCompositeMap;
+	for (auto unit : units)
+	{
+		auto it = mergeMap.find(unit->name());
+		auto compositeName = it != mergeMap.end() ? *it : unit->name();
 
-	DeclarationVisitorSource sourceVisitor;
-	auto dir2 = std::unique_ptr<Export::SourceDir>( sourceVisitor.visitProject(project) );
-	auto map2 = Export::Exporter::exportToFileSystem(pathToProjectContainerDirectory + "/Sources", dir2.get(), &layout);
-	exportMaps().insert(project, map2);
+		auto cIt = nameToCompositeMap.find(compositeName);
+		if (cIt != nameToCompositeMap.end())
+			// case A: the composite that unit is a part of already exists => merge
+			(*cIt)->addUnit(unit);
+		else
+		{
+			// case B: the composite that unit is a part of does not yet exist
+			auto composite = new CodeComposite(compositeName);
+			composite->addUnit(unit);
+			nameToCompositeMap.insert(composite->name(), composite);
+		}
+	}
 
-	return headerVisitor.errors();
+	return nameToCompositeMap.values();
 }
 
-void CppExporter::exportFragment(Export::SourceFragment* fragment)
+void CppExporter::calculateSourceFragments(QList<CodeComposite*> codeComposites)
 {
-	auto dir = new Export::SourceDir(nullptr, "src");
-	auto file = &dir->file("exported.h");
-	file->append(fragment);
-	auto layout = layouter();
-	Export::Exporter::exportToFileSystem("", dir, &layout);
+	DeclarationVisitorHeader headerVisitor;
+	DeclarationVisitorSource sourceVisitor;
+
+	for (auto codeComposite : codeComposites)
+	{
+		for (auto unit : codeComposite->units())
+			if (auto classs = DCast<OOModel::Class>(unit->node()))
+			{
+				unit->headerPart()->setSourceFragment(headerVisitor.visitTopLevelClass(classs));
+				unit->sourcePart()->setSourceFragment(sourceVisitor.visitTopLevelClass(classs));
+			}
+			else
+				Q_ASSERT(false); // TODO: handle non class units
+	}
 }
 
-void CppExporter::calculateSourceFragments(CodeComposite* codeComposite)
+void CppExporter::calculateDependencies(QList<CodeComposite*> codeComposites)
 {
-	DeclarationVisitorHeader visitor;
-	for (auto unit : codeComposite->units())
-		if (auto classs = DCast<OOModel::Class>(unit->node()))
-			unit->setSourceFragment(visitor.visitTopLevelClass(classs));
-		else
-			Q_ASSERT(false);
+	QList<CodeUnitPart*> allHeaderParts;
+	for (auto codeComposite : codeComposites)
+		for (auto unit : codeComposite->units())
+			allHeaderParts.append(unit->headerPart());
+
+	for (auto codeComposite : codeComposites)
+		for (CodeUnit* unit : codeComposite->units())
```

Comment:
Like the previous method, this loop should just call `CodeComposite::calculateDependencies` which in turn should call `CodeUnit::calculateDependencies`.